### PR TITLE
OCD-4175 - Track and analyze (g)(10) Service Based URL availability

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestResultService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        List<SyntheticsAPITestResultShort> testResults = new ArrayList<SyntheticsAPITestResultShort>();
+        try {
+            response = apiProvider.getApiInstance().getAPITestLatestResults(publicTestKey, params);
+            while (response.getResults().size() > 1) {
+                testResults.addAll(response.getResults());
+                Long ts = getMostRecentTimestamp(response.getResults());
+                params.fromTs(ts);
+                response = apiProvider.getApiInstance().getAPITestLatestResults(publicTestKey, params);
+            }
+            LOGGER.info("Found {} tests for monitor {} for {} between {} and {}", testResults.size(), publicTestKey, startTime, endTime);
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestResultService.java]: I read five `{}` values in the string, but only four being passed in. Am I missing something?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestResultService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        } catch (ApiException e) {
+            response = null;
+            LOGGER.error("Could not retrieve results for test key: {}", publicTestKey, e);
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestResultService.java]: Here there's only one `{}`, but two things being passed in. Maybe I just don't understand the `LOGGER`, but the discrepancy seems odd

File: chpl/chpl-resources/src/main/resources/email.properties [hunk 3]
```diff
@@ -354,10 +354,10 @@ developer.missingAttestationChangeRequest.report.salutation=Hello %s,
 developer.missingAttestationChangeRequest.report.paragraph=Emails were sent to the following developers:
 
 #URL Uptime Report
-urlUptime.report.filename=URL Uptime
-urlUptime.report.subject=CHPL URL Uptime Report
-urlUptime.report.heading=CHPL URL Uptime Report
-urlUptime.report.paragraph1.body=See attached file
+serviceBasedUrlUptime.report.filename=Service Based URL Uptime
```
Comment (kekey1) [inline chpl/chpl-resources/src/main/resources/email.properties]: I see the text all around "Service Based URL" but in the UI and in the data model we call it "Service Base URL List". I just want to make sure our terminology is how we want it. Should it be the past-tense Based vs Base? I thought it was like the "base" URL ("Base" as in the foundational URL for all Services) although I truly do not know what is correct.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestService.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    private DatadogSyntheticTestApiProvider apiProvider;
+    private Boolean datadogIsReadOnly;
+    private String datadogTestStartTime;
+    private String datadogTestEndTime;
+    private Long datadogCheckEveryMinutes;
+    private Integer datadogTestTimeout;
+    private String datadogTestLocation;
+
+    public DatadogSyntheticTestService(DatadogSyntheticTestApiProvider apiProvider,
+            @Value("${datadog.syntheticTest.readOnly}") Boolean datadogIsReadOnly,
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestService.java]: Maybe another nitpicky naming thing.. I sometimes see "synthetics" plural and here in the properties it's "synthetic". I think I mostly hear us say the plural one when speaking so I'm wondering if that applies to the property names here, too.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestService.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        this.apiProvider = apiProvider;
+        this.datadogIsReadOnly = datadogIsReadOnly;
+        this.datadogTestStartTime = datadogTestStartTime;
+        this.datadogTestEndTime = datadogTestEndTime;
+        this.datadogCheckEveryMinutes = datadogCheckEveryMinutes;
+        this.datadogTestTimeout = datadogTestTimeout;
+        this.datadogTestLocation = datadogTestLocation;
+    }
+
+    public List<SyntheticsTestDetails> getAllSyntheticTests() {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestService.java]: Here the method name is singular, but a few lines below I see an object called SyntheticsAPITest

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-resources/src/main/resources/email.properties (hunk 3 above)
Comment (tmy1313) [reply chpl/chpl-resources/src/main/resources/email.properties]: image
I based my naming on the ticket description and the wording that I hear Jim using.

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]
There's a checkstyle alert that we should either fix, or update our checkstyle to ignore

File: chpl/chpl-resources/src/main/resources/jobs.xml [hunk 6]
```diff
@@ -843,7 +843,7 @@
         <job>
             <name>Service Base URL List Uptime Report</name>
             <group>chplJobs</group>
-            <description>Sends the Service Based URL Uptime Report</description>
+            <description>Sends the Service Based URL List Uptime Report</description>
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/jobs.xml]:
```suggestion
            <description>Sends the Service Base URL List Uptime Report</description>
```

**Review by andlar — CHANGES_REQUESTED** [review]
There's also a checkstyle error in the jobs.xml file. We should either fix the error, or remove the check

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-resources/src/main/resources/jobs.xml
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/jobs.xml:1]: I've commented on this checkstyle thing a couple of times. Maybe this comment will be more visible. I'm okay going to QA with this, but we really do need to decide if we're going to listen to checkstyle or not. If we are, then we need to fix these issues; if we're not then we need to update our ruleset

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-resources/src/main/resources/jobs.xml
Comment (andlar) [reply chpl/chpl-resources/src/main/resources/jobs.xml:1]: Still has checkstyle issues

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**